choreolib-example/src/main/java/frc/robot/Robot.java

```diff
@@ -0,0 +1,209 @@
+          new Rotation2d()
+        ),
+        new Transform2d(
+          new Translation2d(
+            -kRobotGeometry.FRAME_WIDTH / 2.0,
+            0.0
+          ),
+          new Rotation2d()
+        )
+      )
+    );
+    pickedupNote.onTrue(intake.setNoteCondition(true));
+
+    autoChooser = new ChoreoAutoChooser(
+      Choreo.createAutoFactory(
+        swerve,
+        swerve::getPose,
+        choreoSwerveController(),
+        chassisSpeeds -> swerve.drive(chassisSpeeds, false),
+        () -> DriverStation.getAlliance().orElse(Alliance.Blue).equals(Alliance.Red)
+      )
+    );
+
+    var routines = new AutoRoutines(swerve, intake, shooter);
+
+    autoChooser.addAutoRoutine("Shoot and Backup", routines::shootAndBackup);
+    autoChooser.addAutoRoutine("Four Piece Sub", routines::fourpiece);
+    autoChooser.addAutoRoutine("Five Piece Sub", routines::fivepiece);
+
+    autoChooser.publish();
+  }
+
+  @Override
+  public void robotInit() {}
+
+  @Override
+  public void robotPeriodic() {
+    CommandScheduler.getInstance().run();
+  }
+
+  @Override
+  public void autonomousInit() {
+    CommandScheduler.getInstance().cancelAll();
+    // kinda sketchy, this robot example isnt the best architected robot ive made :skull:
+    intake.setNoteCondition(true).initialize();
+  }
+
+  @Override
+  public void autonomousPeriodic() {
+    autoChooser.pollSelectedAutoRoutine();
+  }
+
+  @Override
+  public void teleopInit() {}
+
+  @Override
+  public void teleopPeriodic() {}
+
+  @Override
+  public void disabledInit() {
+    CommandScheduler.getInstance().cancelAll();
+  }
+
+  @Override
+  public void disabledPeriodic() {}
+
+  @Override
+  public void testInit() {}
+
+  @Override
+  public void testPeriodic() {}
+
+  @Override
+  public void simulationInit() {}
+
+  @Override
+  public void simulationPeriodic() {}
+
+  private static final StructEntry<Pose2d> desiredPoseEntry;
+  private static final StructEntry<Pose2d> currentPoseEntry;
+  private static final StructEntry<ChassisSpeeds> desiredPIDSpeeds;
+  private static final StructEntry<ChassisSpeeds> desiredSpeeds;
+  private static final StructEntry<ChassisSpeeds> currentSpeeds;
+  private static final DoubleEntry timestamp;
+
+  static {
+    var table = NetworkTableInstance.getDefault().getTable("choreo");
+    desiredPoseEntry = table.getStructTopic("DesiredPoseEntry", Pose2d.struct).getEntry(new Pose2d());
+    currentPoseEntry = table.getStructTopic("CurrentPoseEntry", Pose2d.struct).getEntry(new Pose2d());
+    desiredSpeeds = table.getStructTopic("DesiredSpeeds", ChassisSpeeds.struct).getEntry(new ChassisSpeeds());
+    desiredPIDSpeeds = table.getStructTopic("DesiredPIDSpeeds", ChassisSpeeds.struct).getEntry(new ChassisSpeeds());
+    currentSpeeds = table.getStructTopic("CurrentSpeeds", ChassisSpeeds.struct).getEntry(new ChassisSpeeds());
+    timestamp = table.getDoubleTopic("Timestamp").getEntry(0);
+
+    desiredPoseEntry.set(desiredPoseEntry.get());
```

Comment:
What's the point of this section?

Reply:
That was just some debug stuff trying to find a bug I made (I did timer reset instead of restart in the auto traj :/)